Week3/HOMEWORK/js-exercises/project-creditCardValidator.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+console.log('JS1-W3-PROJECT-Credit Card Validator');
+
+function validateCreditCardNumber(cardNumber) {
+  // Credit card number must be 16 digits.
+  if (cardNumber.length !== 16) {
+    return `Error: INVALID Card Number! Your credit card number must be 16 digits!`;
+  }
+
+  // All of the credit card digits must be numbers.
+  let numbers = [];
+  if (typeof cardNumber === 'number') {
+    numbers = ('' + cardNumber).split('');
+  } else if (typeof cardNumber === 'string') {
+    numbers = cardNumber.split('');
+  } else {
+    // else
+    return `Error: INVALID Card Number!`;
+  }
+
+  for (const character of numbers) {
+    if (isNaN(character)) {
+      return `Error: INVALID Card Number! Your credit card number digits must be numbers!`;
+    }
+  }
+
+  // All of the digits cannot be the same number.
+  // And the sum of all the digits must be greater than 16.
+  let digits = [];
+  for (let i = 0; i < 10; ++i) {
+    digits.push(0);
+  }
+
+  let sumOfAllDigits = 0;
+  for (const character of numbers) {
+    const number = +character;
+    digits[number] += 1;
+    sumOfAllDigits += number;
```

Comment:
How about using `reduce` here which we got to know about in the previous question.

Reply:
Thank you for showing other ways. I made this HW in JS1 and we haven't learned reduce yet at that time. We learned this week and it is a really short way...